test/e2e/interconnections/create_test.go

```diff
@@ -0,0 +1,108 @@
+package interconnections
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	"github.com/equinix/metal-cli/internal/interconnections"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/equinix/metal-cli/test/helper"
+
+	metal "github.com/equinix/equinix-sdk-go/services/metalv1"
+	"github.com/spf13/cobra"
+)
+
+func TestInterconnections_Create(t *testing.T) {
+	subCommand := "interconnections"
+	rootClient := root.NewClient(helper.ConsumerToken, helper.URL, helper.Version)
+	randomString := helper.GenerateRandomString(5)
+
+	project := helper.CreateTestProject(t, "metal-cli-interconnections-create-"+randomString)
+	vlan := helper.CreateTestVLAN(t, project.GetId())
+
+	apiClient := helper.TestClient()
+
+	tests := []struct {
+		name    string
+		cmd     *cobra.Command
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "create shared vlan interconnection",
+			cmd:  interconnections.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				connName := "conn-1-" + randomString
+
+				root.SetArgs([]string{subCommand, "create", "-p", project.GetId(), "--vlan", fmt.Sprintf("%d", vlan.GetVxlan()), "-n", connName, "-m", vlan.GetMetroCode(), "-r", "primary", "-t", "shared", "-T", "a_side", "-s", "50000000"})
+
+				out := helper.ExecuteAndCaptureOutput(t, root)
+
+				// Need to find the current user's default org
+				// as orgId is not populated in project. Its created using default org
+				user, _, err := apiClient.UsersApi.
+					FindCurrentUser(context.Background()).
+					Include([]string{"default_organization_id"}).
+					Execute()
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				conns, _, err := apiClient.InterconnectionsApi.
+					OrganizationListInterconnections(context.Background(), user.GetDefaultOrganizationId()).
+					Execute()
+				if err != nil {
+					t.Fatal(err)
+				}
+				if len(conns.GetInterconnections()) < 1 {
+					t.Fatal("Interconnections Not Found. Failed to create Interconnections")
+				}
+
+				var conn *metal.Interconnection
+				for index, c := range conns.GetInterconnections() {
+					if c.GetName() == connName {
+						conn = &conns.GetInterconnections()[index]
+						break
+					}
+				}
+
+				t.Cleanup(func() {
+					helper.CleanupInterconnection(t, conn.GetId())
```

Comment:
Test is failing on this line: https://github.com/equinix/metal-cli/actions/runs/7574424576/job/20634890241#step:5:115

```
=== RUN   TestInterconnections_Create/create_shared_vlan_interconnection
    create_test.go:75: Error when calling `InterconnectionsApi.DeleteVirtualCircuit`` for cbf4bea9-9442-4ac6-a0d7-35c5d0d4e8b4: 422 Unprocessable Entity
--- FAIL: TestInterconnections_Create (3.20s)
    --- FAIL: TestInterconnections_Create/create_shared_vlan_interconnection (1.97s)
FAIL
```

I _think_ this is expected, because it looks like the helper will try to delete the virtual circuits for the interconnection, but the virtual circuits on a shared interconnection are managed by Metal and cannot be deleted by the user.